src/pymodaq/dashboard.py

```diff
@@ -755,24 +755,27 @@ def set_file_preset(self, filename):
 
-                            if ind_plugin == 0:  # should be a master type plugin
-                                if plugin['status'] != "Master":
-                                    logger.error('error in the master/slave type for plugin {}'.format(plug_name))
-                                if plug_init:
-                                    actuators_modules[-1].init_hardware_ui()
-                                    QtWidgets.QApplication.processEvents()
-                                    self.poll_init(actuators_modules[-1])
-                                    QtWidgets.QApplication.processEvents()
-                                    master_controller = actuators_modules[-1].controller
-                            else:
-                                if plugin['status'] != "Slave":
-                                    logger.error('error in the master/slave type for plugin {}'.format(plug_name))
-                                if plug_init:
-                                    actuators_modules[-1].controller = master_controller
-                                    actuators_modules[-1].init_hardware_ui()
-                                    QtWidgets.QApplication.processEvents()
-                                    self.poll_init(actuators_modules[-1])
-                                    QtWidgets.QApplication.processEvents()
+                            try:
+                                if ind_plugin == 0:  # should be a master type plugin
+                                    if plugin['status'] != "Master":
+                                        logger.error('error in the master/slave type for plugin {}'.format(plug_name))
+                                    if plug_init:
+                                        actuators_modules[-1].init_hardware_ui()
+                                        QtWidgets.QApplication.processEvents()
+                                        self.poll_init(actuators_modules[-1])
+                                        QtWidgets.QApplication.processEvents()
+                                        master_controller = actuators_modules[-1].controller
+                                else:
+                                    if plugin['status'] != "Slave":
+                                        logger.error('error in the master/slave type for plugin {}'.format(plug_name))
+                                    if plug_init:
+                                        actuators_modules[-1].controller = master_controller
+                                        actuators_modules[-1].init_hardware_ui()
+                                        QtWidgets.QApplication.processEvents()
+                                        self.poll_init(actuators_modules[-1])
+                                        QtWidgets.QApplication.processEvents()
+                            except Exception as e:
+                                logger.exception(str(e))
```

Comment:
but if there is an error here one should actually show a message to the user, maybe a popup stating that no master has been configured yet (or initiliazed) and that the preset should be reviewed...

Maybe log this specific error and raise again an ActuatorError that would be catched by the next try statement, closing the splashscreen and quitting the application

Reply:
For proper handling I opened issue #175, as I did not know, what your intention is.

Therefore this simple solution.

If it is fine with you, I'll raise that error, which is handled better, and in #175 a proper solution is discussed and implemented.

Reply:
ok let's do that

Reply:
that's fine with me. We'll do another PR from issue #175